Sources/WireGuardKitGo/router.go

```diff
@@ -210,6 +204,12 @@ func (r *Router) Read(bufs []byte, offset int) (n int, err error) {
 	packet := batch.packet
 
 	copy(bufs[offset:], packet)
+	// important to unblock the underlying reader.
+	select {
+	case _, _ = <-r.read.rxShutdown:
+		return 0, io.EOF
+	case batch.completion <- batch:
```

Comment:
Can `batch.completion <- batch` block?

Reply:
It can, but as long as the read worker goroutine can proceed (it is not waiting on anything else in between), this is/should be fine.
<!-- Reviewable comment -ODfvAoC-oTsTkuP3CbH:byu0clg -->
<!-- Sent from Reviewable.io -->